src/main/java/randoop/main/GenTests.java

```diff
@@ -1335,4 +1358,150 @@ private Path getResourceDirectoryPath(String resourceDirectory) {
+          addConstructorParameterTypesIfShould(constructor, dependenciesClassnames, accessibility);
+        }
+      } catch (ClassNotFoundException e) {
+        throw new RandoopUsageError(String.format("Cannot load class %s", classname));
+      }
+    }
+    return dependenciesClassnames;
+  }
+
+  /**
+   * Returns names of classes methods in methodlist depend on. Does not add dependencies of methods
+   * that should be omitted. Does not add classes that are not accessible or should be omitted.
+   *
+   * @param accessibilityPredicate an accessibility predicate
+   * @return classnames of dependencies
+   */
+  public static Set<@ClassGetName String> getDependentClassnamesFromMethodList(
+      AccessibilityPredicate accessibilityPredicate) {
+    Set<@ClassGetName String> classnames = new TreeSet<>();
+    if (methodlist == null) {
+      return Collections.emptySet();
+    }
+
+    ReflectionPredicate reflectionPredicate = new DefaultReflectionPredicate();
+    try (EntryReader reader = new EntryReader(methodlist, "(//|#).*$", null)) {
+      for (String signature : reader) {
+        if (!shouldOmitMethod(signature)) {
+          AccessibleObject accessibleObject = null;
+          try {
+            accessibleObject =
+                SignatureParser.parse(signature, accessibilityPredicate, reflectionPredicate);
+          } catch (SignatureParseException | FailedPredicateException e) {
+            continue;
+          }
+          if (accessibleObject instanceof Constructor) {
+            addConstructorParameterTypesIfShould(
+                (Constructor<?>) accessibleObject, classnames, accessibilityPredicate);
+          }
+          if (accessibleObject instanceof Method) {
+            addMethodParameterTypesIfShould(
+                (Method) accessibleObject, classnames, accessibilityPredicate);
+          }
+        }
+      }
+    } catch (IOException e) {
+      throw new RandoopUsageError("Can`t read methods from " + methodlist.toString());
+    }
+    return classnames;
+  }
+
+  /**
+   * Tests whether a method should be omitted. Returns true if the signature matches any pattern in
+   * omit_method list, false otherwise.
+   *
+   * @param signature signature of method to test
+   * @return true if method should be ommited
+   */
+  private static boolean shouldOmitMethod(String signature) {
+    for (Pattern pattern : omit_methods) {
+      if (pattern.matcher(signature).find()) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Adds parameter types of method to collection if method is accessible by accessibility predicate
+   * and parameter type is not String, primitive, should not be omitted and is accessible by
+   * accessibility predicate
+   *
+   * @param method method
+   * @param classnames collection to add parameter types to
+   * @param accessibilityPredicate accessibility predicate
+   */
+  private static void addMethodParameterTypesIfShould(
```

Comment:
When https://github.com/randoop/randoop/pull/939 is merged, you will be able to have just one method `addParameterTypesIfShould` rather than being forced to have two overloads, one for methods and one for constructors.